### PR TITLE
fix: Stringify the kubeletExtraArgs values for AL2/Ubuntu

### LINKED ISSUE #3818: kubelet-eks failed to start due to unexpected token '<'
### Version

__Karpenter Version:__ v0.27.3

__Kubernetes Version:__ v1.24 EKS on region `us-west-2`

launch to the instance directly and check logs
```sh
$ sudo snap logs kubelet-eks
2023-04-27T06:31:13Z systemd[1]: Stopped Service for snap application kubelet-eks.daemon.
2023-04-27T06:31:13Z systemd[1]: Started Service for snap application kubelet-eks.daemon.
2023-04-27T06:31:13Z kubelet-eks.daemon[8680]: /snap/kubelet-eks/104/wrappers/run-with-config-args: array assign: line 7: syntax error near unexpected token `<'
2023-04-27T06:31:13Z systemd[1]: snap.kubelet-eks.daemon.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
2023-04-27T06:31:13Z systemd[1]: snap.kubelet-eks.daemon.service: Failed with result 'exit-code'.
2023-04-27T06:31:13Z systemd[1]: snap.kubelet-eks.daemon.service: Scheduled restart job, restart counter is at 5.
2023-04-27T06:31:13Z systemd[1]: Stopped Service for snap application kubelet-eks.daemon.
2023-04-27T06:31:13Z systemd[1]: snap.kubelet-eks.daemon.service: Start request repeated too quickly.
2023-04-27T06:31:13Z systemd[1]: snap.kubelet-eks.daemon.service: Failed with result 'exit-code'.
2023-04-27T06:31:13Z systemd[1]: Failed to start Service for snap application kubelet-eks.daemon.
```

i checked the launch template user data, and find it might be caused of not escape the char `<`
```
--kubelet-extra-args '--node-labels=karpenter.sh/capacity-type=on-demand,karpenter.sh/provisioner-name=default,tests=biganimal --register-with-taints=biganimal=true:NoSchedule --max-pods=20 --pods-per-core=2 --system-reserved=cpu=100m,ephemeral-storage=1Gi,memory=100Mi --kube-reserved=ephemeral-storage=3Gi,memory=100Mi,cpu=200m --eviction-hard=nodefs.inodesFree<10%,memory.available<5%,nodefs.available<10% --eviction-soft=nodefs.inodesFree<15%,memory.available<500Mi,nodefs.available<15% --eviction-soft-grace-period=nodefs.available=1m30s,nodefs.inodesFree=2m0s,memory.available=1m0s --eviction-max-pod-grace-period=60 --image-gc-high-threshold=85 --image-gc-low-threshold=80 --cpu-cfs-quota=true'
```

after i remove these arguments with `<`, it works fine

### Expected Behavior

`kubelet-eks` could start successfully

### Actual Behavior

node get stuck in the notReady status
```
$ kubectl get nodes
NAME                                           STATUS     ROLES    AGE    VERSION
ip-192-168-127-95.us-west-2.compute.internal   NotReady   <none>   7m7s
ip-192-168-48-97.us-west-2.compute.internal    Ready      <none>   18h    v1.24.11-eks-a59e1f0
ip-192-168-70-109.us-west-2.compute.internal   Ready      <none>   18h    v1.24.11-eks-a59e1f0
```

### Steps to Reproduce the Problem

n/a

### Resource Specs and Logs

deploy the provisioner with following configuration:
```yaml
  # Karpenter provides the ability to specify a few additional Kubelet args.
  # These are all optional and provide support for additional customization and use cases.
  kubeletConfiguration:
    clusterDNS: ["10.0.1.100"]
    containerRuntime: containerd
    systemReserved:
      cpu: 100m
      memory: 100Mi
      ephemeral-storage: 1Gi
    kubeReserved:
      cpu: 200m
      memory: 100Mi
      ephemeral-storage: 3Gi
    evictionHard:
      memory.available: 5%
      nodefs.available: 10%
      nodefs.inodesFree: 10%
    evictionSoft:
      memory.available: 500Mi
      nodefs.available: 15%
      nodefs.inodesFree: 15%
    evictionSoftGracePeriod:
      memory.available: 1m
      nodefs.available: 1m30s
      nodefs.inodesFree: 2m
    evictionMaxPodGracePeriod: 60
    imageGCHighThresholdPercent: 85
    imageGCLowThresholdPercent: 80
    cpuCFSQuota: true
    podsPerCore: 2
    maxPods: 20
```

deploy the awstemplate with the following configuration:
```yaml
  amiFamily: Ubuntu
  amiSelector:
    aws-ids: ami-0d429719359d44d16
```

```sh
$ sudo snap logs kubelet-eks
2023-04-27T06:31:13Z systemd[1]: Stopped Service for snap application kubelet-eks.daemon.
2023-04-27T06:31:13Z systemd[1]: Started Service for snap application kubelet-eks.daemon.
2023-04-27T06:31:13Z kubelet-eks.daemon[8680]: /snap/kubelet-eks/104/wrappers/run-with-config-args: array assign: line 7: syntax error near unexpected token `<'
2023-04-27T06:31:13Z systemd[1]: snap.kubelet-eks.daemon.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
2023-04-27T06:31:13Z systemd[1]: snap.kubelet-eks.daemon.service: Failed with result 'exit-code'.
2023-04-27T06:31:13Z systemd[1]: snap.kubelet-eks.daemon.service: Scheduled restart job, restart counter is at 5.
2023-04-27T06:31:13Z systemd[1]: Stopped Service for snap application kubelet-eks.daemon.
2023-04-27T06:31:13Z systemd[1]: snap.kubelet-eks.daemon.service: Start request repeated too quickly.
2023-04-27T06:31:13Z systemd[1]: snap.kubelet-eks.daemon.service: Failed with result 'exit-code'.
2023-04-27T06:31:13Z systemd[1]: Failed to start Service for snap application kubelet-eks.daemon.
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #3818

**Description**

Fixes a bug in the Ubuntu AmiFamily where any `kubeletConfiguration` that generated kubelet arguments with the `<` symbol would not be properly parsed by the AMIFamily's bootstrap script, thus causing the Node never to join. This PR fixes this bug by wrapping all string values in the kubelet arguments with double-quotes so that these values will be treated as strings when parsed.

**How was this change tested?**

* `FOCUS=KubeletConfiguration make e2etests`
* `make presubmit`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Comment by jonathan-innis** [comment]
All tests pass
:white_check_mark: utilization-pr-3841-69fvj tests passed
:white_check_mark: ipv6-pr-3841-ljqjj tests passed
:white_check_mark: drift-pr-3841-2lb5r tests passed
:white_check_mark: interruption-pr-3841-n64hz tests passed
:white_check_mark: consolidation-pr-3841-ld27l tests passed
:white_check_mark: chaos-pr-3841-gk6tx tests passed
:white_check_mark: machine-pr-3841-q6nsf tests passed
:white_check_mark: integration-pr-3841-97mhn tests passed
:white_check_mark: upgrade-pr-3841-n6rsv tests passed

File: pkg/providers/amifamily/bootstrap/eksbootstrap.go [hunk 1]
```diff
@@ -65,86 +64,90 @@ func (e EKS) eksBootstrapScript() string {
...
-		userData.WriteString(fmt.Sprintf(" \\\n--container-runtime %s", e.ContainerRuntime))
-	}
-	if kubeletExtraArgsStr := strings.Trim(kubeletExtraArgs.String(), " "); len(kubeletExtraArgsStr) > 0 {
-		userData.WriteString(fmt.Sprintf(" \\\n--kubelet-extra-args '%s'", kubeletExtraArgsStr))
-	}
-	if e.KubeletConfig != nil && len(e.KubeletConfig.ClusterDNS) > 0 {
-		userData.WriteString(fmt.Sprintf(" \\\n--dns-cluster-ip '%s'", e.KubeletConfig.ClusterDNS[0]))
-	}
-	return userData.String()
+	return lo.Filter(args, func(s string, _ int) bool { return s != "" })
```
Comment (bwagner5) [inline pkg/providers/amifamily/bootstrap/eksbootstrap.go:123]:
```suggestion
	return lo.Compact(args)
```

File: pkg/providers/amifamily/bootstrap/eksbootstrap.go [hunk 2]
```diff
@@ -65,86 +64,90 @@ func (e EKS) eksBootstrapScript() string {
...
-	if e.KubeletConfig != nil && e.KubeletConfig.PodsPerCore != nil {
-		kubeletExtraArgs.WriteString(fmt.Sprintf(" --pods-per-core=%d", ptr.Int32Value(e.KubeletConfig.PodsPerCore)))
+	if args := e.kubeletExtraArgs(); len(args) > 0 {
+		userData.WriteString(fmt.Sprintf(" \\\n--kubelet-extra-args '%s'", strings.Join(args, " ")))
 	}
+	return userData.String()
+}
 
+func (e EKS) kubeletExtraArgs() (args []string) {
+	args = append(args, e.nodeLabelArg(), e.nodeTaintArg())
```
Comment (bwagner5) [inline pkg/providers/amifamily/bootstrap/eksbootstrap.go:93]: can we check and bail if `e.KubeletConfig == nil` at the top  so we can get rid of all the reoccurring checks? (have to shift e.AWSENILimitedPodDensity into that check too, but still seems cleaner)

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by bwagner5 — DISMISSED** [review]
lgtm

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by bwagner5 — APPROVED** [review]
LGTM

**Pull request merged**